packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -4364,6 +4364,203 @@ void main() {
+    expect(controller.text, equals(testText), reason: 'on $platform');
+
+    // Select All
+    await sendKeys(
+      tester,
+      <LogicalKeyboardKey>[
+        LogicalKeyboardKey.keyA,
+      ],
+      shortcutModifier: true,
+      platform: platform,
+    );
+
+    expect(
+      selection!,
+      equals(
+        const TextSelection(
+          baseOffset: 0,
+          extentOffset: testText.length,
+          affinity: TextAffinity.upstream,
+        ),
+      ),
+      reason: 'on $platform',
+    );
+    expect(controller.text, equals(testText), reason: 'on $platform');
+
+    // Cut
+    await sendKeys(
+      tester,
+      <LogicalKeyboardKey>[
+        LogicalKeyboardKey.keyX,
+      ],
+      shortcutModifier: true,
+      platform: platform,
+    );
+
+    expect(
+      selection!,
+      equals(
+        const TextSelection(
+          baseOffset: 0,
+          extentOffset: testText.length,
+          affinity: TextAffinity.upstream,
+        ),
+      ),
+      reason: 'on $platform',
+    );
+    expect(controller.text, equals(testText), reason: 'on $platform');
+    expect(
+      (await Clipboard.getData(Clipboard.kTextPlain))!.text,
+      equals(clipboardContent),
+      reason: 'on $platform',
+    );
+
+    // Copy
+    await sendKeys(
+      tester,
+      <LogicalKeyboardKey>[
+        LogicalKeyboardKey.keyC,
+      ],
+      shortcutModifier: true,
+      platform: platform,
+    );
+
+    expect(
+      selection!,
+      equals(
+        const TextSelection(
+          baseOffset: 0,
+          extentOffset: testText.length,
+          affinity: TextAffinity.upstream,
+        ),
+      ),
+      reason: 'on $platform',
+    );
+    expect(controller.text, equals(testText), reason: 'on $platform');
+    expect(
+      (await Clipboard.getData(Clipboard.kTextPlain))!.text,
+      equals(testText),
+      reason: 'on $platform',
+    );
+
+    // Delete
+    await sendKeys(
+      tester,
+      <LogicalKeyboardKey>[
+        LogicalKeyboardKey.delete,
+      ],
+      platform: platform,
+    );
+    expect(
+      selection!,
+      equals(
+        const TextSelection(
+          baseOffset: 0,
+          extentOffset: testText.length,
+          affinity: TextAffinity.upstream,
+        ),
+      ),
+      reason: 'on $platform',
+    );
+    expect(controller.text, equals(testText), reason: 'on $platform');
+
+    // Backspace
+    await sendKeys(
+      tester,
+      <LogicalKeyboardKey>[
+        LogicalKeyboardKey.backspace,
+      ],
+      platform: platform,
+    );
+    expect(
+      selection!,
+      equals(
+        const TextSelection(
+          baseOffset: 0,
+          extentOffset: testText.length,
+          affinity: TextAffinity.upstream,
+        ),
+      ),
+      reason: 'on $platform',
+    );
+    expect(controller.text, equals(testText), reason: 'on $platform');
+  }
+
+  testWidgets('keyboard shortcuts respect read-only on android', (WidgetTester tester) async {
```

Comment:
Instead of doing it like this, make one `testWidgets` test with a `TargetPlatformVariant` `variant:` argument:

```dart
  testWidgets('keyboard shortcuts respect read-only', (WidgetTester tester) async {
	String platform = describeEnum(defaultTargetPlatform).toLowerCase();
	/// Insert the contents of testReadOnlyShortcuts here.
  },
  skip: kIsWeb,
  variant: TargetPlatformVariant.all());
```

It'll run the test once for each platform, with a different description including the platform.

Reply:
oh, nice. thanks! it was a shameless copy from somewhere above :)

Reply:
Yeah, we should probably also fix the ones above too (you don't have to).